python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py

```diff
@@ -335,11 +344,249 @@ def write_parquet_file(context):
         self.execute_queries(sql_queries)
 
 
+class SnowflakeResource(ConfigurableResourceFactory):
+    """A resource for connecting to the Snowflake data warehouse. The returned resource object is an
+    instance of :py:class:`SnowflakeConnection`.
+
+    A simple example of loading data into Snowflake and subsequently querying that data is shown below:
+
+    Examples:
+        .. code-block:: python
+
+            from dagster import job, op
+            from dagster_snowflake import SnowflakeResource
+
+            @op
+            def get_one(snowflake_resource: Resource[SnowflakeConnection]):
+                snowflake_resource.execute_query('SELECT 1')
+
+            @job(resource_defs={
+                'snowflake_resource': SnowflakeResource(
+                    account=EnvVar("SNOWFLAKE_ACCOUNT"),
+                    user=EnvVar("SNOWFLAKE_USER"),
+                    password=EnvVar("SNOWFLAKE_PASSWORD")
+                    database="MY_DATABASE",
+                    schema="MY_SCHEMA",
+                    warehouse="MY_WAREHOUSE"
+                )}
+            )
+            def my_snowflake_job():
+                get_one()
+
+            my_snowflake_job.execute_in_process()
```

Comment:
We need to make a decision about how strongly we will push for the using resource_defs on  `Definitions` versus arguments on `job` versus passing to `execute_in_process`. We are kind of all over the place.

Reply:
has there been any discussion on this front?